test/e2e/gatekeeper_controller_test.go

```diff
@@ -123,8 +127,217 @@ var _ = Describe("Gatekeeper", func() {
 
 			return apierrors.IsNotFound(err)
 		}, deleteTimeout, pollInterval).Should(BeTrue())
+
+		By("Clean config", func() {
+			Eventually(func() bool {
+				err := K8sClient.Get(ctx, types.NamespacedName{
+					Name:      "config",
+					Namespace: gatekeeperNamespace,
+				}, &gkv1alpha1.Config{})
+				if err == nil {
+					return false
+				}
+
+				return apierrors.IsNotFound(err)
+			}, deleteTimeout, pollInterval).Should(BeTrue())
+		})
 	})
 
+	Describe("Test config resource", Ordered, func() {
+		// The default exempt namespaces
+		defaultNamespaces := []wildcard.Wildcard{
+			"kube-*", "multicluster-engine",
+			"hypershift", "hive", "rhacs-operator", "open-cluster-*", "openshift-*",
+		}
+
+		AfterEach(func() {
+			By("Clean gatekeeper")
+			_, err := KubectlWithOutput("delete", "gatekeeper", "gatekeeper", "--ignore-not-found")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("Clean config surely", func() {
+				Eventually(func() bool {
+					err := K8sClient.Get(ctx, types.NamespacedName{
+						Name:      "config",
+						Namespace: gatekeeperNamespace,
+					}, &gkv1alpha1.Config{})
+					if err == nil {
+						return false
+					}
+
+					return apierrors.IsNotFound(err)
+				}, deleteTimeout, pollInterval).Should(BeTrue())
+
+				Consistently(func() bool {
+					err := K8sClient.Get(ctx, types.NamespacedName{
+						Name:      "config",
+						Namespace: gatekeeperNamespace,
+					}, &gkv1alpha1.Config{})
+					if err == nil {
+						return false
+					}
+
+					return apierrors.IsNotFound(err)
+				}, 5, pollInterval).Should(BeTrue())
```

Comment:
Is 5 seconds really long enough? I'm wondering if this `Consistently` can be removed.